#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  basicTimeStep 4
}
Viewpoint {
  orientation 0.14511351659890084 0.1263531731949892 -0.9813138860343604 1.4958987504604058
  position -1.1326142979135523 3.070829849599254 0.8111316172268287
}
TexturedBackground {
  skyColor [
    0 0 0
    0 0 0
    0 0 0
  ]
}
TexturedBackgroundLight {
}
Floor {
  appearance Parquetry {
    type "light strip"
    textureTransform TextureTransform {
      scale 0.2 0.2
    }
  }
}
Robot {
  translation 0 0 0.1
  children [
    InertialUnit {
      translation 0 0 0.07
      name "imu"
    }
    DEF body Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0 0
        }
      }
      geometry Box {
        size 0.06 0.1 0.15
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.07 -0.06
      }
      device [
        PositionSensor {
          name "encoder_left"
        }
        RotationalMotor {
          name "motor_left"
        }
      ]
      endPoint DEF wheel1 Solid {
        translation 0 0.07 -0.06
        rotation 1 -8.439231465925498e-16 -5.88783590645965e-17 1.570799999999996
        children [
          DEF wheelshape Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.8 0.4 0.5
              }
            }
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheelshape
        physics DEF wheel_physics Physics {
          density -1
          mass 0.08
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 -0.07 -0.06
      }
      device [
        PositionSensor {
          name "encoder_right"
        }
        RotationalMotor {
          name "motor_right"
        }
      ]
      endPoint Solid {
        translation 0 -0.07 -0.06
        rotation 1 -6.672880693987596e-16 3.9252239376397623e-16 1.5707999999999982
        children [
          DEF wheelshape Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0.8 0.4 0.5
              }
            }
            geometry Cylinder {
              height 0.03
              radius 0.04
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheelshape
        physics DEF wheel_physics Physics {
          density -1
          mass 0.08
        }
      }
    }
  ]
  boundingObject USE body
  physics Physics {
    density -1
    mass 2
  }
  controller "forward"
  window "generic_window"
}
